lib/validator.ex

```diff
@@ -79,7 +80,7 @@ defmodule UltraDark.Validator do
   end
 
   defp appropriate_coinbase_output?([coinbase | transactions], block_index) do
-    if Block.total_block_fees(transactions) + Block.calculate_block_reward(block_index) == List.first(coinbase.outputs).amount, do: :ok, else: {:error, "Coinbase output is invalid"}
+    if D.equal?(D.add(Block.total_block_fees(transactions), Block.calculate_block_reward(block_index)), List.first(coinbase.outputs).amount), do: :ok, else: {:error, "Coinbase output is invalid"}
```

Comment:
since we're nesting this logic inside of a function call now, it might make sense to set a new variable `total_fees = Block.total_block_fees(transactions)`, and using that in the comparison, just to make the code more readable